cmd/promtool/tsdb.go

```diff
@@ -398,24 +398,28 @@ func openBlock(path, blockID string) (*tsdb.DBReadOnly, tsdb.BlockReader, error)
 	if err != nil {
 		return nil, nil, err
 	}
-	blocks, err := db.Blocks()
-	if err != nil {
-		return nil, nil, err
-	}
+
 	var block tsdb.BlockReader
 	if blockID != "" {
-		for _, b := range blocks {
-			if b.Meta().ULID.String() == blockID {
-				block = b
-				break
-			}
+		b, err := db.Block(nil, filepath.Join(path, blockID))
+		if err != nil {
+			return nil, nil, err
+		}
+		block = b
+	} else {
+		blocks, err := db.Blocks()
```

Comment:
How about we add db.LastBlock() function here because we are only checking that block here?

Reply:
Thank you for your review!

> How about we add db.LastBlock() function here because we are only checking that block here?

That sounds nice!
Exactly we need not to read all blocks.
So I'll try it!